### PR TITLE
Calculate batch and transaction hashes in sov-app-template

### LINKED ISSUE #250: Calculate batch and transaction hashes in sov-app-template::apply_blob

### PR DESCRIPTION
**Pull request by bkolad**
close #250

### CODE REVIEW

File: module-system/sov-app-template/src/lib.rs [hunk 1]
```diff
@@ -202,13 +206,33 @@ where
...
+struct BatchDataAndHash {
+    hash: [u8; 32],
+    data: Vec<u8>,
+}
+
+impl BatchDataAndHash {
+    fn new<C: Context>(batch: impl Buf) -> BatchDataAndHash {
+        let mut reader = batch.reader();
+        let mut batch_data = Vec::new();
+        reader
```
Comment (bkolad) [inline module-system/sov-app-template/src/lib.rs:225]: The only way I found to calculate the batch hash was by converting it to a slice of bytes.

File: module-system/sov-app-template/src/lib.rs [hunk 2]
```diff
@@ -120,8 +122,10 @@ where
         let mut tx_receipts = Vec::with_capacity(txs.len());
 
         // Process transactions in a loop, commit changes after every step of the loop.
-        for tx in txs {
+        for (tx, raw_tx_hash) in txs {
             batch_workspace = batch_workspace.to_revertable();
+
+            //let tx_hash = tx.hash();
```
Comment (preston-evans98) [inline module-system/sov-app-template/src/lib.rs:128]: nit: let's remove the commented code

**Review by preston-evans98 — APPROVED** [review]
This looks great overall. Since all that we need these hashes for is indexing receipts (which is currently not part of the rollup state), we can skip this hashing in the zk context. No need to do that in this PR, but let's open an issue so we don't forget

File: module-system/sov-app-template/src/lib.rs (hunk 2 above)
Comment (bkolad) [reply module-system/sov-app-template/src/lib.rs]: done

**Pull request merged**